### DISCUSSION TITLE
Integer literal with leading zero (MicroPython vs CPython)

### DISCUSSION THREAD

**Discussion by aivarannamaa** [micropython/micropython#10967] [General]
Python 3 parser does not allow integer literals with leading zero, eg `012`. MicroPython does, but unlike Python 2 it does not treat it as an octal but simply ignores the leading zero.

Is this documented somewhere? Wouldn't it be clearer if MicroPython disallowed leading zero like Python 3?

**Comment by sosi-deadeye** [comment]
Yes, it is true:
```
[andre@andre-Fujitsu-i5 ~]$ micropython
MicroPython v1.19.1 on 2023-02-16; linux [GCC 12.2.1] version
Use Ctrl-D to exit, Ctrl-E for paste mode
>>> int("011")
11
>>> int("00000000000011")
11
>>>
```

Look carefully at the documentation: https://docs.python.org/3/library/functions.html#int
> Base **0** also **disallows leading zeros**: **int('010', 0) is not legal**, while int('010') and int('010', 8) are.

**Reply by aivarannamaa** [reply]
Your example is about `int` function, but my question is about MicroPython parser, which differs from Python 3 parser regarding leading zero in integer literals.

```python
MicroPython v1.19 on 2022-07-27; Pimoroni Tiny 2040 with RP2040
Type "help()" for more information.
>>> 011
11
```

vs.

```python
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 011
  File "<stdin>", line 1
    011
    ^
SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers
```

vs. 

```python
Python 2.7.18 (default, Jul  1 2022, 10:30:50) 
[GCC 11.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> 011
9
```

As you see, MicroPython follows neither Python 3 nor Python 2. This could be a potential confusion.
